spec/apps/load-error/load-error.spec.js

```diff
@@ -0,0 +1,37 @@
+import * as singleSpa from "single-spa";
+
+describe(`load-error`, () => {
+  it(`will mount an application before loading another application finishes`, async () => {
+    singleSpa.registerApplication(
+      "load-error",
+      () => Promise.reject(Error(`load failed`)),
+      location => location.hash.startsWith("#load-error")
+    );
+    location.hash = "#load-error";
+    await singleSpa.triggerAppChange();
+    expect(singleSpa.getAppStatus("load-error")).toBe(singleSpa.LOAD_ERROR);
+  });
+
+  it(`puts app into LOAD_ERROR status before firing error event`, async () => {
+    let numErrs = 0;
+
+    await singleSpa.triggerAppChange();
+    singleSpa.addErrorHandler(handleError);
```

Comment:
Why do you need to trigger an app change and add an error handler before an app is registered?

Reply:
I wanted to make sure that single-spa isn't in the middle of an update (maybe from another test) when adding the error handler, so that the only error that could possibly be handled is the one caused later on in this test.

Reply:
👍 that was my guess, but figured I'd make sure it wasn't accidental